src/assets/Playwright/Playwright-debugging.md

```diff
@@ -0,0 +1,129 @@
+ 
+```js
+await chromium.launch({ headless: false, slowMo: 100 }); // or firefox, WebKit
+```
+ 
+ 
+## Playwright Inspector
+Playwright comes with a default GUI tool that we can use to inspect our scripts. Through this tool, you'll be able to step-over each script and evaluate them in real-time. There are a few ways through which we can open Playwright inspector.
+ 
+### 1. PWDEBUG
+Playwright provides us with an environment variable that'll configure it in debugging mode and opens the inspector. Set the `PWDEBUG` variable to `1` or `console`.
+ 
+In my case, I'm running the e2e test in debug mode by setting `PWDEBUG=1` .
+```bash
+PWDEBUG=1 npm run test:e2e tests/acceptance/features/todo.feature
+```
+ 
+This will open up an inspector like so
+ 
+![Playwright Inspector](/src/assets/Playwright/images/playwright-inspector.png "Playwright Inspector")
+ 
+As you can see this gives me the test scripts, now I can either step over each script or run everything at once
+ 
+ 
+**step over each script**
+![Playwright Inspector](/src/assets/Playwright/images/step-over.png "Step over script")
+ 
+**run all at once**
+ 
+![Playwright Inspector](/src/assets/Playwright/images/run-all.png "run-all script")
+ 
+With each step-over, the inspector will step through each line of the test highlighting the selector as you go. You can also see the logs that display each action that the tests perform.
+ 
+![Playwright Inspector](/src/assets/Playwright/images/trace_viewer2.png "highlight selector and logs")
+ 
+Additionally, you'll also be able to access the browser developers' tools.
+ 
+ 
+### 2. page.pause()
+The next method to launch the inspector is to use `page.pause()` in the script.
+ 
+```js
+When('the user adds {string} to the todo list using the webUI',async function (item) {
+   // fill the item that was input from the feature file, to the inputText field in the UI
+   await page.fill(todoInput , item)
+   // pauses the test execution and launches Playwright inspector
+   await page.pause()
+   // click the button
+   await page.click(todoButton)
+})
+ 
+```
+ 
+This will pause the execution of the test and launch Playwright inspector right before clicking the button.
+ 
+![Playwright Inspector](/src/assets/Playwright/images/pause.png "page.pause")
+ 
+Now, we can perform similar operations as explained above.
+ 
+ 
+## Trace Viewer
+Trace viewer is another GUI tool to explore the recorded Playwright traces of the tests after the tests have been executed. This is especially essential while running the tests in the Continuous Integration (CI) environment.
+ 
+Let's see how we can set it up in our end-to-end tests. In the `cucumber.conf.js` file inside `before hook` we can add the following configuration
+ 
+```js
+Before(async function () {
+   global.context = await global.browser.newContext();
+   // start tracing the test execution by enabling the screenshots and snapshots
+   await context.tracing.start({ screenshots: true, snapshots: true });
```

Comment:
```suggestion
   await global.context.tracing.start({ screenshots: true, snapshots: true });
```

Reply:
works fine even without `global` but no harm in adding it either